cdmtaskservice/mongo.py

```diff
@@ -135,7 +136,52 @@ async def get_job(
             raise NoSuchJobError(f"No job with ID '{job_id}' exists")
         # TODO PERF build up the job piece by piece to skip S3 path validation
         return models.AdminJobDetails(**doc) if as_admin else models.Job(**doc)
+
+    async def _update_job_state(
+        self,
+        job_id: str,
+        current_state: models.JobState,
+        state: models.JobState,
+        time: datetime.datetime,
+        push: dict[str, Any] | None = None,
+    ):
+        res = await self._col_jobs.update_one(
+            {
+                models.FLD_JOB_ID: _require_string(job_id, "job_id"),
```

Comment:
works. But it might be clearer to validate these parameters first and store them in local variables instead of calling validators inline.

Reply:
yeah, I used to do it that way, but you wind up with a potentially long list of validators at the start of each method. Inlining them makes the method easier to read imo since the validator calls are so simple